Task/#48 - documentation improvements, especially for governing Questionnaires

ArdonToonstra opened the pull request:
Closes #48

ArdonToonstra committed:
Update ACP data exchange methods and conformance

Expanded documentation to specify four transaction methods for exchanging ACP information using RESTful API, covering both R4 and STU3 FHIR versions. Added definitions for ACP Actor Consulter and Provider, and clarified conformance requirements for each actor to improve interoperability guidance.

ArdonToonstra committed:
Clarify rationale for form-based QuestionnaireResponse

Added explanation for including the form-based QuestionnaireResponse method, highlighting its benefits for data providers and noting the increased effort required from ACP Actor Consulters.

ArdonToonstra committed:
Update ACP Actor Provider requirements

Removed recommendation for providers to support multiple transaction groups, focusing on preference for individual resources method over form-based method for system-to-system exchange.

ArdonToonstra committed:
Merge branch 'develop' into feature/#45

ArdonToonstra committed:
Fixed uri to static ACP dataset in ArtDecor

ArdonToonstra committed:
Remove incorrect version from address.country

ArdonToonstra committed:
Merge pull request #49 from IKNL/feature/#45

Feature/#45

ArdonToonstra committed:
Add patient bundle and Postman collection generators

Introduces two utility scripts: one for generating FHIR patient bundle resources by grouping related files, and another for creating a Postman collection with PUT and validation requests for FHIR resources. Both scripts scan the fsh-generated/resources directory, exclude certain resource types, and output results to the util folder.

ArdonToonstra committed:
#53 fix typo in search url

ArdonToonstra committed:
Remove performer.actor constraint and fix instance title

Deleted the performer.actor reference constraint from the ACPProcedure profile and corrected a typo in the instance title for F1-ACP-Procedure-01-10-2020.

ArdonToonstra committed:
Update documentation for questionnaire resource workflow

Expanded .github/copilot-instructions.md with detailed guidance on questionnaire management, metadata best practices, and directory structure. Added input/resources/README.md to document the process for creating and adjusting Questionnaire and QuestionnaireResponse resources. Minor update to validator/README.md to clarify validator scope. Added image resource for documentation.

ArdonToonstra committed:
Update questionnaire governance docs and remove image

Expanded and clarified the README for questionnaire and QuestionnaireResponse governance, detailing the process for exporting, adjusting, and registering questionnaires and responses. Removed outdated questionnaire notes from decisions.md and deleted the unused image.png file.

ArdonToonstra committed:
Remove fsh-generated directory from version control

Deleted all files under fsh-generated/ and updated .gitignore to exclude the directory. This prevents generated FSH content from being tracked in the repository.

ArdonToonstra committed:
Enable SUSHI in Firely Terminal workflow

Set SUSHI_ENABLED to true in the r4_firely_terminal GitHub Actions workflow to activate SUSHI processing during the CI pipeline.

ArdonToonstra committed:
Remove generated FSH index and reference files

Deleted fsh-generated/data/fsh-index.json, fsh-generated/fsh-index.txt, and fsh-generated/includes/fsh-link-references.md, which contained generated indexes and link references for FSH resources. This cleanup may be part of a build process change or to avoid tracking generated artifacts in version control.

MariekeMassa commented on input/resources/README.md:
```diff
@@ -0,0 +1,188 @@
+1. Save as `QuestionnaireResponse-[PatientName]-[Date].json` in `input/resources/`
+   - Example: `QuestionnaireResponse-HendrikHartman-20201001.json`
+2. Clean up the exported JSON:
+   - Remove the `meta.tag` field containing `lformsVersion`
```
In the existing QuestionnaireResponses, this tag is still there. The meta.profile is however removed, which is not mentioned here.

MariekeMassa replied:
If it doesn't bother us, I would say we leave it (both the tag and profile), as it indeed saves steps.

ArdonToonstra committed:
Merge pull request #66 from IKNL/fix/#64-procedure

Remove performer.actor constraint and fix instance title

ArdonToonstra committed:
Harmonized mappings and obligations by setting them on the root of extensions

ArdonToonstra committed:
Harmonized mappings and obligations by setting them on the root of extension

ArdonToonstra committed:
Turning off dotnet validation

ArdonToonstra committed:
Try inserting the Obligation extension in the right  place

ArdonToonstra committed:
Add author and source to QR responses #48

Included 'author' and 'source' fields in both QuestionnaireResponse-HendrikHartman-20201001.json and QuestionnaireResponse-HendrikHartman-20221108.json to specify the healthcare professional and patient associated with each response.

ArdonToonstra committed:
bump Q and QR version to latest

ArdonToonstra committed:
Merge branch 'main' into develop

ArdonToonstra committed:
Refactor obligation extension rules for consistency

Replaces custom ObligationRulesInsertWithExtensionLocation rules with a simplified ObligationRules rule using the $obligation alias. Updates all references in FSH files to use the new rule, improving maintainability and consistency.

ArdonToonstra committed:
Added missing obligation rules

Introduces additional obligation rules to set the code to #SHALL:no-error and the actor to ACPActorConsulter for the specified path. This enhances the expressiveness of obligation handling in the RuleSet.

MariekeMassa commented on input/resources/README.md:
```diff
@@ -0,0 +1,188 @@
+2. Clean up the exported JSON:
+   - Remove the `meta.tag` field containing `lformsVersion`
+   - Add proper `subject` and `author` references
+   - Ensure `questionnaire` canonical URL matches your Questionnaire
```
So then we first need to be sure of the definitive canonical and then manually change it here?

ArdonToonstra committed:
Add script to strip standards status extensions (HACK until proper solution is implemented)

Introduces util/strip_standards_status_extensions.py to remove specific FHIR standards status extensions from JSON and normative badges from HTML files. Updates _genonce.bat to run this script after IG Publisher execution, improving resource validation and publication workflows.

ArdonToonstra committed:
Merge branch 'develop' into fix/#50-sushi-obligation-extension-handling

ArdonToonstra committed:
Merge pull request #69 from IKNL/fix/#50-sushi-obligation-extension-handling

Fix/#50 sushi obligation extension handling

Removes all fsh-generated resource from source control
Improves Obligation extension FSH syntax -- better and more robust inserting
Harmonizes the mappings and obligations. Always on the root of an extension for example. This improves the rendering in the IG publisher
Adds a HACKY workaround to remove the N labels in the IG publisher output for now until either SUSHI if fixed, or the Snapshot generator and we can depend on a new zib profiles. https://chat.fhir.org/#narrow/channel/215610-shorthand/topic/.E2.9C.94.20Issue.20adding.20extension.20when.20base.20has.20extensions.20in.20snap.2E.2E.2E/with/561905267

ArdonToonstra committed:
Merge branch 'develop' into task/#48

ArdonToonstra committed:
Update instructions for exporting QuestionnaireResponse JSON

Revised step 4 to enhance exported JSON by adding an 'id' field and clarified instructions for including subject, author, questionnaire URL, status, and authored date. Removed outdated guidance about cleaning up the 'meta.tag' field.

HarmkeKoning reviewed (approved):
According to Marieke this PR can be approved
